src/Microsoft.DotNet.MSBuildSdkResolver/VSSettings.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+#if NET46
+using Microsoft.VisualStudio.Setup.Configuration;
+#endif
+
+namespace Microsoft.DotNet.MSBuildSdkResolver
+{
+    internal sealed class VSSettings
+    {
+        private readonly object _lock = new object();
+        private readonly string _settingsFilePath;
+        private readonly bool _disallowPrereleaseByDefault;
+        private FileInfo _settingsFile;
+        private bool _disallowPrerelease;
+
+        public VSSettings()
+        {
+#if NET46
+            if (!Interop.RunningOnWindows)
+            {
+                return;
+            }
+
+            var configuration = new SetupConfiguration();
+            var instance = configuration?.GetInstanceForCurrentProcess();
```

Comment:
Need try/catch around the interop here. Getting COMException when running an msbuild.exe outside of VS folder.